cmd/erasure-metadata.go

```diff
@@ -158,11 +158,11 @@ func (fi FileInfo) ToObjectInfo(bucket, object string, versioned bool) ObjectInf
 
 	// Update storage class
 	if fi.TransitionTier != "" {
-		objInfo.StorageClass = fi.TransitionTier
+		objInfo.StorageClass = globalAPIConfig.storageClass(fi.TransitionTier)
```

Comment:
We should leave the storage class unmodified in this layer since FileInfo can be converted to ObjectInfo for internal uses too.